smartsim/_core/_cli/cli.py

```diff
@@ -73,14 +74,34 @@ def execute(self, cli_args: t.List[str]) -> int:
             self.parser.print_help()
             return 0
 
-        app_args = cli_args[1:]
-        self.args = self.parser.parse_args(app_args)
+        app_args = cli_args[1:]  # exclude the path to executable
+        subcommand = cli_args[1]  # first positional arg is the subcommand
 
-        if not (menu_item := self.menu.get(app_args[0], None)):
+        menu_item = self.menu.get(subcommand, None)
+        if not menu_item:
             self.parser.print_help()
             return 0
 
-        return menu_item.handler(self.args)
+        if menu_item.is_plugin:
+            self.args, unparsed_args = self.parser.parse_known_args(app_args)
+        else:
+            unparsed_args = []
+            self.args = self.parser.parse_args(app_args)
+
+        return menu_item.handler(self.args, unparsed_args)
+
+    def _register_plugin(self, item: MenuItemConfig) -> None:
+        parser = self.subparsers.add_parser(
+            item.command, description=item.description, help=item.description
+        )
+        if item.configurator:
+            item.configurator(parser)
+
+        self.menu[item.command] = item
```

Comment:
Can add a check to the nature of:
```py
if item.commnad in self.menu:
    raise ValueError("Cannot overwrite existing CLI command")
```
here to make sure a plugin does not accidentally overwrite another plugin (or worse a first class command)?

Reply:
Fantastic idea. Done!